### DISCUSSION TITLE
How can I use the translation module only?

### DISCUSSION THREAD

**Discussion by rspreet92** [openai/whisper#201] [General]
I want to skip audio detection part and use it for translation only, how can I do that?

**Comment by lmmx** [comment]
If you pass the `--language` flag and a language code you’ll skip the language detection, and just go straight into the task (whether that’s translate or transcribe).

**Reply by flesnuk** [reply]
I think he could also refer to use whisper as machine translation only, for text-to-text. Not involving the audio. I'm not sure, but that's what I understood from reading the OP.

**Reply by rspreet92** [reply]
yes exactly, sorry I couldn't phrase it well, I want to use it for text-to-text translation.

**Reply by jianfch** [reply]
since you can't do it directly, hacky way to do it would be to feed text-to-speech output into whisper.

**Comment by jongwook** [comment]
(Similar to #36) the model doesn't have a separate translation module and performs speech translation end-to-end. So it's not directly possible to use it as a text translation model, although it could be an interesting research project to distill a text translation model out of Whisper.

**Comment by ManilBen** [comment]
What if I want to translate one audio to multiple target languages, is it possible to use some intermediate representation of the input text and perform translation by using it as input to the translation part, instead of having to pass the same input audio multiple times?

**Reply by phineas-pta** [reply]
whisper translate best from any lang TO english, translate to any other lang may not work as expected
